core/types/data.go

```diff
@@ -238,36 +239,43 @@ func (s DataSourceSpecSigners) String() string {
 }
 
 type DataSourceSpecPropertyKey struct {
-	Name string
-	Type DataSourceSpecPropertyKeyType
+	Name                string
+	Type                DataSourceSpecPropertyKeyType
+	NumberDecimalPlaces *uint64
 }
 
 func (k DataSourceSpecPropertyKey) String() string {
 	return fmt.Sprintf(
-		"name(%s) type(%s)",
+		"name(%s) type(%s) decimals(%s)",
 		k.Name,
 		k.Type.String(),
+		strconv.FormatUint(*k.NumberDecimalPlaces, 10),
 	)
 }
 
 func (k DataSourceSpecPropertyKey) IntoProto() *datapb.PropertyKey {
-	return &datapb.PropertyKey{
-		Name: k.Name,
-		Type: k.Type,
+	pk := &datapb.PropertyKey{
+		Name:                k.Name,
+		Type:                k.Type,
+		NumberDecimalPlaces: k.NumberDecimalPlaces,
 	}
+
+	return pk
 }
 
 func (k *DataSourceSpecPropertyKey) DeepClone() *DataSourceSpecPropertyKey {
 	return &DataSourceSpecPropertyKey{
-		Name: k.Name,
-		Type: k.Type,
+		Name:                k.Name,
+		Type:                k.Type,
+		NumberDecimalPlaces: k.NumberDecimalPlaces,
```

Comment:
Even if we should keep this field, this is not a deep clone. the clone would share this pointer with the original object

Reply:
Addressed - the two values has the same address when its done like this -> https://github.com/vegaprotocol/vega/pull/6935/commits/db7f4cd7157e7edeeb1c9c464b9094521d5be6fb#diff-38be7115fe6ae7d810706c7f1c7224e82ecc7da720c0e330d82f869b1e7a7189L261-L265
Thanks!